diff --git a/CMakeLists.txt b/CMakeLists.txt
index d114cac..5b3fc58 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,7 +28,7 @@ message(STATUS "Found Eigen3 Version: ${Eigen3_VERSION} Path: ${Eigen3_DIR}")
 # -------------
 
 add_library(Spectra INTERFACE)
-add_library(spectra::spectra ALIAS spectra)
+add_library(Spectra::Spectra ALIAS Spectra)
 target_include_directories(Spectra
 		INTERFACE
 		$<INSTALL_INTERFACE:include>
